api/v1alpha1/datadogagent_default.go

```diff
@@ -407,8 +424,13 @@ func DefaultDatadogAgent(ad *DatadogAgent) *DatadogAgent {
 		}
 	}
 
-	if defaultedAD.Spec.Features != nil && defaultedAD.Spec.Features.OrchestratorExplorer != nil {
-		defaultedAD.Spec.Features.OrchestratorExplorer = DefaultDatadogFeatureOrchestratorExplorer(defaultedAD.Spec.Features.OrchestratorExplorer)
+	if defaultedAD.Spec.Features != nil {
+		if defaultedAD.Spec.Features.OrchestratorExplorer != nil {
+			defaultedAD.Spec.Features.OrchestratorExplorer = DefaultDatadogFeatureOrchestratorExplorer(defaultedAD.Spec.Features.OrchestratorExplorer)
+		}
```

Comment:
same here the if is maybe not needed because you already handle the case in the function

Reply:
So for this one, I actually want to keep it that way. 
If the `spec.Features` is != nil, and the orchestrator explorer is not activated, I don't think we should mutate the field.